### PR TITLE
Complete python notes

### PR DESCRIPTION
**Pull request by Abdiramen**
I've added the sections you were missing and made a few minor changes to in spacing.

### CODE REVIEW

File: python/python.md [hunk 1]
```diff
@@ -118,17 +118,17 @@ x = 5
     - Works the same as a [list](#list)
 
 #### Dictionary
-+ Iterable (iterates ove rkeys by default
++ Iterable (iterates over keys by default)
 + type is `dict`
 + Maps keys to values
 + Keys and values can vary in type
-    - Keys `MUST` be hashable
+    - Keys `MUST` be hashtable
```
Comment (kevinschoonover) [inline python/python.md]: hashable.

File: python/python.md [hunk 2]
```diff
@@ -365,9 +365,322 @@ x = 5
...
+    ```python
+    def print_name(person):
+        try:
+            print(person["name"])
+        except KeyError:
+            print("Person has no name")
+    ```
+
+#### More detail about Exception
++ All exceptions are instances of classes that derive form `BaseExceptin`
```
Comment (kevinschoonover) [inline python/python.md]: BaseException

File: python/python.md [hunk 3]
```diff
@@ -365,9 +365,322 @@ x = 5
...
++ Most exception you'll use derived from`Exception`
```
Comment (kevinschoonover) [inline python/python.md]: from `Exception`

File: python/python.md [hunk 4]
```diff
@@ -365,9 +365,322 @@ x = 5
...
++ Example
+    ```python
+    # You can keep the caught exception as a variable;
+    # useful for error messages and debugging
+    try:
+        f = open("file.txt")
+    except FileNotFoundError as e:
+        print("Caught:", e)
+
+    # You cna catch several types of Exception with on
```
Comment (kevinschoonover) [inline python/python.md]: can

File: python/python.md [hunk 5]
```diff
@@ -365,9 +365,322 @@ x = 5
...
+    - Example
+        ```python
+        def func(a, b, *args, d=10, e=11, **kwargs):
+            pass
+        ```
+
+## Comprehensions, Expressions, and Generators
+### List Comprehension!!!
+```python
+    [int(x) for x in["10","20"]] # list created immidiatly
```
Comment (kevinschoonover) [inline python/python.md]: immediately

**Comment by Abdiramen** [comment]
Glad it all worked out!

**Pull request merged**